integration-cli/docker_cli_run_test.go

```diff
@@ -4300,3 +4300,62 @@ func (s *DockerSuite) TestRunTooLongHostname(c *check.C) {
 	hostname3 := "this-is-a-hostname-with-64-bytes-so-will-not-give-an-error.local"
 	dockerCmd(c, "run", "--hostname", hostname3, "busybox", "echo", "test")
 }
+
+func (s *DockerSuite) TestRunRmWithRestartPolicy(c *check.C) {
```

Comment:
Self comment: This test case is failing of existing bug, fix is already pushed: https://github.com/docker/docker/pull/21839